RoddyCore/src/de/dkfz/roddy/config/FilenamePatternHelper.groovy

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2017 eilslabs.
+ *
+ * Distributed under the MIT License (license terms are at https://www.github.com/eilslabs/Roddy/LICENSE.txt).
+ */
+
+package de.dkfz.roddy.config
+
+import groovy.transform.CompileStatic
+
+import static de.dkfz.roddy.StringConstants.BRACE_RIGHT
+import static de.dkfz.roddy.StringConstants.EMPTY
+import static de.dkfz.roddy.StringConstants.EMPTY
+import static de.dkfz.roddy.StringConstants.SPLIT_COMMA
+import static de.dkfz.roddy.StringConstants.SPLIT_EQUALS
+
+/**
+ * Created by heinold on 10.01.17.
+ */
+@CompileStatic
+class FilenamePatternHelper {
+
+    public static class Command {
+        public final String name;
+        public final Map<String, CommandAttribute> attributes = new HashMap<String, CommandAttribute>();
+
+        public Command(String name, Map<String, CommandAttribute> attributes) {
+            this.name = name;
+            if (attributes != null)
+                this.attributes.putAll(attributes);
+        }
+    }
+
+    public static class CommandAttribute {
+        public final String name;
+        public final String value;
+
+        public CommandAttribute(String name, String value) {
+            this.name = name;
+            this.value = value;
+        }
+    }
+
+    public static Command extractCommand(String commandID, String temp, int startIndex = -1) {
+        if (startIndex == -1)
+            startIndex = temp.indexOf(commandID);
+        int endIndex = temp.indexOf(BRACE_RIGHT, startIndex);
+        String command = temp[startIndex .. endIndex]
+
+        Map<String, CommandAttribute> attributes = [:]
+
+        String[] split = command.split(SPLIT_COMMA);
+        for (int i = 1; i < split.length; i++) { //Start with the first option.
+            String _split = split[i].replaceAll("[^0-9a-zA-Z=.-_+#]", EMPTY);
+            String[] attributeSplit = _split.split(SPLIT_EQUALS);
```

Comment:
it wasn't at the beginning. It replaces everything EXCEPT the stuff in the square brackets.